File: src/main/java/ru/olejkai/task_vsr/search/TaskSearchValues.java
package ru.olejkai.task_vsr.search;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.swing.*;
import java.sql.Date;
import java.sql.Timestamp;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class TaskSearchValues {
    //Parameter of search
    String employeeIdTasker;
    String taskProblemId;
    Timestamp dateBegin;
    String employeeIdExecuter;
    String departmentIdExecuter;
    Timestamp dataFinish;
    Integer status;
    //pageable
    private Integer pageNumber;
    private Integer pageSize;
    //sorting
    private String sortColumn;
    private String sortDirection;
}

File: frontend/src/app/models/TaskSearchValues.ts



export class TaskSearchValues{

  //Parameter of search
  employeeIdTasker?:number;
  taskProblemId?:number;
  dateBegin?:Date;
  employeeIdExecuter?:number;
  departmentIdExecuter?:number;
  dataFinish?:Date;
  status?:number;
  //pageable
  pageNumber?:number;
  pageSize?:number;
  //sorting
  sortColumn?:string;
  sortDirection?:string;


  constructor(employeeIdTasker: number, taskProblemId: number, dateBegin: Date, employeeIdExecuter: number, departmentIdExecuter: number, dataFinish: Date, status: number, pageNumber: number, pageSize: number, sortColumn: string, sortDirection: string) {
    this.employeeIdTasker = employeeIdTasker;
    this.taskProblemId = taskProblemId;
    this.dateBegin = dateBegin;
    this.employeeIdExecuter = employeeIdExecuter;
    this.departmentIdExecuter = departmentIdExecuter;
    this.dataFinish = dataFinish;
    this.status = status;
    this.pageNumber = pageNumber;
    this.pageSize = pageSize;
    this.sortColumn = sortColumn;
    this.sortDirection = sortDirection;
  }
}
